opengrok-indexer/src/test/java/org/opengrok/indexer/history/PerforceRepositoryTest.java

```diff
@@ -105,4 +106,25 @@ public void testHistoryAndAnnotations() throws Exception {
             }
         }
     }
+
+    @Test
+    public void testProtectFilename() throws Exception {
+        PerforceRepository instance = new PerforceRepository();
+        ArrayList<SimpleImmutableEntry<String,String>>   testmap = new ArrayList<SimpleImmutableEntry<String,String>>() {
+            {
+                add(new SimpleImmutableEntry<>("Testfile 34", "Testfile 34"));
+                add(new SimpleImmutableEntry<>("Test%52", "Test%2552"));
```

Comment:
maybe add another test case with multiple occurrences of chars to be replaced (tests `relplaceAll()`)